lib/crud.js

```diff
@@ -33,14 +33,64 @@ function load(id, force, cb) {
   });
 }
 
-function remove(id, cb) {
-  const q = ["UPDATE entity",
-    "SET entity_removed = now()",
-    "WHERE entity_id = $1 AND entity_removed IS NULL"];
-  pgClient.query(q.join(" "), [id], (err, res) => {
+function remove(id, correlationId, cb) {
+  if (typeof correlationId === "function") {
+    cb = correlationId;
+    correlationId = null;
+  }
+
+  load(id, (err, doc) => {
+    if (err) return cb(err);
+    if (!doc) return cb(new Error("no such entity"));
+
+    const emptyDoc = {
+      id: id,
+      type: doc.type,
+      meta: {correlationId: correlationId}
+    };
+
+    return upsert(emptyDoc, (upsertErr) => {
+      if (upsertErr) return cb(upsertErr);
+      const q = [
+        "UPDATE entity",
+        "SET entity_removed = now()",
+        "WHERE entity_id = $1 AND entity_removed IS NULL"];
```

Comment:
ta bort villkoret för entity_removed is null tycker jag, finns randfall där man p.g.a. fel, manuell uppdatering etc. kan ha ett värde där (och uppdateringen i entity_version är ju redan gjord så vi vill sätta now() alltid)

Reply:
Åtgärdad.